math/minuit2/src/Numerical2PGradientCalculator.cxx

```diff
@@ -225,11 +196,16 @@ FunctionGradient Numerical2PGradientCalculator::operator()(const MinimumParamete
    mpiproc.SyncVector(gstep);
 #endif
 
-#ifdef DEBUG
-   std::cout << "Calculated Gradient at x =   " << par.Vec() << std::endl;
-   std::cout << "fcn(x) = " << fcnmin << std::endl;
-   std::cout << "Computed gradient in N2PGC " << grd << std::endl;
-#endif
+   // print after parallel processing to avoid synchronization issues
+   print.Debug([&](std::ostream& os) {
```

Comment:
This prints the message in a single line in a no readable message. 
What about changing to something like this that is more readable: 
```
  print.Debug([&](std::ostream& os) {
      const int pr = os.precision(13);
      os << std::endl;
      os << std::setw(14) << "Parameter" << std::setw(14) << "Gradient" << std::setw(14) << "g2 "
         << std::setw(14) << "step" << std::endl;
      for (int i = 0; i < int(n); i++) {
         const int iext = Trafo().ExtOfInt(i);
         os << std::setw(14) << Trafo().Name(iext) << " " << grd(i) << " " << g2(i) << " " << gstep(i) << std::endl;
      }
      os.precision(pr);
   });
```

similar could be done for other messages (see for example print.Debug above)

Reply:
Ok. I have no strong opinion on how messages should be formatted, I always assumed that the formatting will be changed in follow-up PRs. Because we have the filters now, there is no reason left to make messages one-liners.

Reply:
OK, I agree. Let's do the formatting in following PR's. But anyway a `std::endl` was missing

Reply:
I implemented the change as you suggested.